### ISSUE TITLE
After upgrade from 4.8 to 5.0 RAW files from Fujifilm GFX 100 / 100S cameras are all purple after import.

### ISSUE THREAD

**Issue by tin-** [darktable-org/darktable#18073]
### Describe the bug

I've been using previous release 4.8 to process RAF files from Fujifilm GFX and it worked as expected.
After fresh installation of darktable 5.0 (Windows 10 PC) and import of GFX files images show up normal in the lighttable library, but after import to darkroom for editing they turn heavy purple with all default settings. Applying color calibration or playing with colorspace does not make it any workable. Turning OpenCL on or off does not change the result. I've tried files from the two cameras - GFX 100 and GFX 100S, and results are identical.

Same issue was reported here as well:
https://discuss.pixls.us/t/purple-images-with-fuji-gfx-in-dt-5-0/47164

### Steps to reproduce

1. Use RAW file from GFX, such as one listed in https://discuss.pixls.us/t/purple-images-with-fuji-gfx-in-dt-5-0/47164
2. Import to darktable 5.0 (windows 10, didn't test on other OS)
3. Open file in darkroom. Image will be heavy purple.

### Expected behavior

Should open with proper colors, as the old 4.8 does.

### Where did you obtain darktable from?

downloaded from www.darktable.org

### darktable version

5.0.0

### What OS are you using?

Windows

### What is the version of your OS?

Windows 10 1709 (16299.125)

### Describe your system?

Intel Core i7-7980XE, 96GB memory, EVGA GeForce 1080 Ti

### Are you using OpenCL GPU in darktable?

Yes

**Comment by jenshannoschwalm** [comment]
@kmilos did you encounter this before? white points wrong? bayer pattern? RGGB / BGGR ?

**Comment by jenshannoschwalm** [comment]
Right. So the initial value could be bad or it's the black points.

**Comment by jenshannoschwalm** [comment]
No chance to get me into rawspeed code and indeed that should not be the problem. Also it seems it's the same issue :-)

**Comment by robgrub** [comment]
I got the same problem, using MacOS Sequoia 15.1.1

**Comment by jenshannoschwalm** [comment]
@kmilos I checked brute-force using libraw for all raf files. works for this camera. For some reason completely failing on other raf files ...

**Comment by SteveLamont** [comment]
I'm experiencing the same issue with RAF images which loaded in a previous (4.6.1) version.
For what it's worth, RawTherapee loads the images fine.
If you'd like sample images from a GFX100S, please let me know and I will provide.

**Comment by jenshannoschwalm** [comment]
@ralfbrown @kmilos i tried to use libraw on all RAF images and was somewhat surprised. The image in question was good, others had a failure like the one mentioned in this issue. We also changed the loader mechanism, any bell ringing?

**Comment by ralfbrown** [comment]
The only thing the changed loader mechanism might be doing is switching whether rawspeed or LibRaw gets used for an image.  There have been cases where different firmware versions result in different data being recorded, and perhaps rawspeed and LibRaw disagree on which way to interprest things....

I just checked, and 5.0.0 and master both show purple for GFX 100 and GFX 100 II samples from RPU (my local copy) and are fine for the other GFX models.  I clearly didn't see that when making the camera styles back in June or so.  BTW, 5.0.0 *is* using rawspeed as loader for both purple images.  Interestingly, when I tried swapping the CFA colors for the GFX 100 in cameras.xml (because it crops an odd number of sensels on top and bottom), darktable told me the file was corrupt as a result of rawspeed throwing an "Unexpected Bayer phase" exception....

**Comment by jenshannoschwalm** [comment]
Yes, I found that too.

**Comment by jenshannoschwalm** [comment]
@kmilos @LebedevRI i don't know the rawspeed internals and it's working at all, but i can confirm:
1. Adding the removed `<Crop x="0" y="4" width="-146" height="-6"/>` lines bring back correct colors.
2. I can't find a problem with odd/even iop/rawspeed croppers, that has and is working correct
3. I would have expected defining it as a BGGR pattern would solve the issue but as @ralfbrown has already reported above this results in an "Unexpected Bayer phase"

**Comment by dllu** [comment]
![image](https://github.com/user-attachments/assets/6b116937-75f4-4a9f-9315-4fe0a5f70d22)

I can confirm that https://github.com/darktable-org/rawspeed/pull/770 fixes it. Thanks @kmilos !

**Comment by jenshannoschwalm** [comment]
Good to me too :-) Checked for (new) failures on all raf sample files i have ...

**Comment by sengebusch** [comment]
f.y.i.: same problem for Fuji GFX 100 ii.

I was not able to try the patch yet as I am using binaries usually.

**Comment by jenshannoschwalm** [comment]
@kmilos did you try that and found it working? For me it didn't and I also think it should **not** as we always can freely change that and the pattern is changed accordingly.

**Comment by sengebusch** [comment]
in dt 4.8 I found  `<Crop x="0" y="0" width="-146" height="-6"/>`  in `cameras.xml` for GFX 100xx cameras.
This lines seems to be missing in dt 5.0 for GFX at all. Do you know, why they were removed?

I downgraded dt on my working horse from 5.0 to 4.8 using a back up, 
but I will install a fresh system in parallel with dt 5.0 this evening and test.

**Comment by jenshannoschwalm** [comment]
@kmilos still hoping for your proper rawspeed fix getting merged for 5.0.1

**Comment by SteveLamont** [comment]
On 1/16/25 06:12, sengebusch wrote:
> in dt 4.8 I found |<Crop x="0" y="0" width="-146" height="-6"/>| in | 
> cameras.xml| for GFX 100xx cameras.

I can confirm that adding that line to cameras.xml, for the GFX 100S, at 
least, produces what appears to my eye to be a properly rendered image.

Perhaps it is as simple as that?

Regards.

							spl

**Comment by paperdigits** [comment]
RPU is back up, btw.

**Comment by CritLoren** [comment]
Would be nice if this kind of continuously updated file was decoupled from the main app, as to not require waiting for a full stable update, using a snapshot, or setting up a temporary fix to get around the issue

**Comment by paperdigits** [comment]
I feel like decoupling the file would be extremely confusing, and the application already sows enough confusion as it is.

**Comment by LebedevRI** [comment]
The `cameras.xml` is, technically, decoupled, but
1. this wouldn't have helped here since a code change was needed too
2. i'm not sure how well this update works, because then the actual decoder version would need to be tracked per-camera entry, and that isn't being done.
